doc/api_ref/bigint.rst

```diff
@@ -1,35 +1,72 @@
 BigInt
 ========================================
 
-``BigInt`` is Botan's implementation of a multiple-precision integer. Thanks to
-C++'s operator overloading features, using ``BigInt`` is often quite similar to
-using a native integer type. The number of functions related to ``BigInt`` is
-quite large, and not all of them are documented here. You can find the complete
-declarations in ``botan/bigint.h`` and ``botan/numthry.h``.
+``BigInt``, in ``bigint.h``, is an implementation of a signed magnitude
+multiple-precision integer, which is used to implement certain older public key
+algorithms such as RSA. It also appears in other contexts, for example X.509
+certificate serial numbers are technically integer values and can be quite
+large, and so are represented using a ``BigInt``.
 
-.. cpp:class:: BigInt
+A ``BigInt`` is a sequence of smaller integers of type ``word``; this type is
+defined to be either ``uint32_t`` or ``uint64_t``, depending on the word size of
+the processor.
 
-   .. cpp:function:: BigInt()
+.. warning::
 
-      Create a BigInt with value zero
+   While it is possible to use the APIs provided by ``BigInt`` as a general
+   calculation facility, it is **extremely inadvisable** that you attempt to
+   implement a cryptographic scheme of any kind directly using ``BigInt``.
+   Botan internally has many facilities for fast and side channel safe
+   arithmetic which are not exposed to callers.
 
-   .. cpp:function:: BigInt::from_u64(uint64_t n)
+   In general, as a library user, avoid doing anything with ``BigInt`` besides
+   serializing or deserializing them as required to call other interfaces.
+   Some of the general calculation facilities of ``BigInt`` may be made internal
+   to the library in a future major release.
 
-      Create a BigInt with value *n*
+.. cpp:class:: BigInt
 
-   .. cpp:function:: BigInt(std::string_view str)
+   .. cpp:function:: static BigInt BigInt::from_string(std::string_view str)
 
       Create a BigInt from a string. By default decimal is expected. With an 0x
-      prefix instead it is treated as hexadecimal. A ``-`` prefix to indicate
+      prefix, instead it is treated as hexadecimal. A ``-`` prefix to indicate
       negative numbers is also accepted.
 
-   .. cpp:function:: BigInt(std::span<const uint8_t> buf)
+   .. cpp:function:: static BigInt::from_bytes(std::span<const uint8_t> buf)
 
       Create a BigInt from a binary array (big-endian encoding).
```

Comment:
Is there a way to control the sign using this function?

Reply:
No, it is always positive. I updated the doc